### ISSUE TITLE
Require try-catch block around await: try-catch-await (Enforce a try-catch statement for all uses of await)

### ISSUE THREAD

**Issue by gkiely** [typescript-eslint/typescript-eslint#5345]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/

### Description

Require a `try` `catch` block around all uses of await.

It is possible to call await (on a fetch call for example) and not handle an error or changed json schema, leading to a runtime error.

It would be good if it was possible to enforce this to ensure developers handle errors.

### Fail

```typescript
const getPostIds = (async () => {
  const res = await fetch('/api/posts');
  const posts = await res.json(); // Server returns {} instead of expected []
  return posts.map(o => o.id); // Uncaught TypeError: {}.map is not a function
});
```

### Pass

```typescript
const getPostIds = (async () => {
  try {
    const res = await fetch('/api/posts');
    const posts = await res.json();
    return posts.map(o => o.id);
  } catch {
    console.log('Error fetching posts');
  }
});
```

### Additional Info

Related: https://github.com/xjamundx/eslint-plugin-promise/blob/development/docs/rules/catch-or-return.md

**Comment by bradzacher** [comment]
For non async code, enforcing `.catch` makes sense because otherwise the errors are lost to the abyss and don't interrupt your program.

For async code however this pattern doesn't make as much sense because any errors will just throw up the await chain.

> It is possible to call await and not handle an error

This isn't unique to async functions though - any function could throw at any time. Just being async doesn't make it any different.

---

In your example the dangerous thing is that you're working with `fetch`, which is untyped. So the runtime values don't align with your types (I.e. Your code is unsafe).

---

As outline above - I don't think this proposal enforces a good practice. Additionally it does not check for any TypeScript-specific features.
For these reasons I am going to reject this proposal.

**Comment by gkiely** [comment]
Understood, thanks for reviewing.
